### PR TITLE
[TT-8693] Support JSVM as a base identity provider in multi-auth

### PR DESCRIPTION
**Pull request by furkansenharputlu**
This PR implements the support of using JSVM as a base identity provider in multi-auth. 
The `base_identity_provided_by` should be set as `custom_auth` and inside javascript code the return object should set `AuthValue` to define id of the new session to be used as a base identity.

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4952/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4688996342)

File: apidef/api_definitions.go [hunk 1]
```diff
@@ -73,6 +73,7 @@ const (
 	// For multi-type auth
 	AuthTypeNone  AuthTypeEnum = ""
 	AuthToken     AuthTypeEnum = "auth_token"
+	Custom        AuthTypeEnum = "custom"
```
Comment (buger) [inline apidef/api_definitions.go:76]: Why it is called custom, and not JSVM or similar?

Comment (furkansenharputlu) [reply apidef/api_definitions.go:76]: Can we enable coprocess auth with python, jsvm or goplugin at the same time? I think no. That's why we can combine them as `custom`. What do you think?

Comment (buger) [reply apidef/api_definitions.go:76]: Can we then call it custom_auth or custom_auth_plugin? 

But also it makes it work only for JSVM right, you will need to do the same logic for coprocess,grpc too?

Comment (furkansenharputlu) [reply apidef/api_definitions.go]: Done.

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4952/merge
Commit: d0ecf2e6bcb1df238d641d9e0232ab34d3f2636b 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4690061580)

**Review by buger — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4952/merge
Commit: e09f5eb2878ccf8ac85c8738ab187983ac8b7173 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4690593232)

**Pull request merged**